ybd/repos.py

```diff
@@ -160,11 +155,26 @@ def mirror_has_ref(gitdir, ref):
         return out == 0
 
 
+def reinit_as_bare(gitdir):
+    with app.chdir(gitdir):
```

Comment:
I'd have `os.path.join(gitdir, f)` the paths from `os.listdir(gitdir)` and passed cwd=gitdir to call, or added gitdir to the list of arguments.